modules/misc/qt.nix

```diff
@@ -127,14 +127,25 @@ in {
   config = let
 
     # Necessary because home.sessionVariables doesn't support mkIf
-    envVars = filterAttrs (n: v: v != null) {
+    envVars = let
+      inherit (config.home) profileDirectory;
+      qtVersions = with pkgs; [ qt5 qt6 ];
+    in filterAttrs (n: v: v != null) {
       QT_QPA_PLATFORMTHEME = if cfg.platformTheme == "gtk" then
         "gtk2"
       else if cfg.platformTheme == "qtct" then
         "qt5ct"
       else
         cfg.platformTheme;
       QT_STYLE_OVERRIDE = cfg.style.name;
+      QT_PLUGIN_PATH = "$QT_PLUGIN_PATH\${QT_PLUGIN_PATH:+:}"
+        + (lib.concatStringsSep ":"
+          (map (qt: "${profileDirectory}/${qt.qtbase.qtPluginPrefix}")
+            qtVersions));
+      QML2_IMPORT_PATH = "$QML2_IMPORT_PATH\${QML2_IMPORT_PATH:+:}"
+        + (lib.concatStringsSep ":"
+          (map (qt: "${profileDirectory}/${qt.qtbase.qtQmlPrefix}")
+            qtVersions));
```

Comment:
Tested in my config:

```
$ echo $QT_PLUGIN_PATH
/nix/store/hdvlnzaxdm06xkipg62mj30aa3fzi5fa-fcitx5-with-addons-5.1.1/lib/qt-6/plugins:/etc/profiles/per-user/thiagoko/lib/qt-5.15.10/plugins:/etc/profiles/per-user/thiagoko/lib/qt-6/plugins
$ echo $QML2_IMPORT_PATH
/etc/profiles/per-user/thiagoko/lib/qt-5.15.10/qml:/etc/profiles/per-user/thiagoko/lib/qt-6/qml
```

Seems to be working fine, including properly keeping the current plugins in PATH.